protocol/messenger_sync_settings.go

```diff
@@ -112,6 +111,14 @@ func (m *Messenger) extractSyncSetting(syncSetting *protobuf.SyncSetting) (*sett
 		value = json.RawMessage(v)
 	}
 
+	if syncSetting.GetType() == protobuf.SyncSetting_DISPLAY_NAME {
+		m.account.Name = value.(string)
+		err = m.multiAccounts.SaveAccount(*m.account)
```

Comment:
@qfrank if the only need for this is to update account's display name in case when we're fetching data from waku, then it's redundant cause it is already handled [here](https://github.com/status-im/status-go/blob/5005b7b3cf45fed801257060fd68e19c4422e317/protocol/messenger_backup_handler.go#L77)

Reply:
emm..but [this](https://github.com/status-im/status-go/blob/5005b7b3cf45fed801257060fd68e19c4422e317/protocol/messenger_backup_handler.go#L77) only happens for back up, what if it's coming from #syncDevices? @saledjenic

Reply:
Yes, that's true, that's why I said "if the only need for this is..." what do you think of moving that part to the place where you need it, [here](https://github.com/status-im/status-go/blob/5005b7b3cf45fed801257060fd68e19c4422e317/protocol/messenger.go#L3711-L3716), after that block or do the handling in a new fuction. 
Cause an intention of `extractSyncSetting` func is to convert `protobuf.SyncSetting` to `settings.SyncSettingField` that's why I wouldn't set account form that function.
What do you think about that?

Reply:
Oh, good point, sorry i forgot u renamed `handleSyncSetting` to `extractSyncSetting`, but i think it would be better not rename it so that we can keep the same code style as before as u can c that there'r lots of handleXXX within handleRetrievedMessages @saledjenic

Reply:
and we should invoke something like messageState.Response.AddSettings() to modify Response(@cammellos told me before in my old prs before), sorry I did not review your pr in time 😥

Reply:
What I wanted initially is to have clear separation among handling device syncing and applying backed up data, like having different handling functions which at the end share the same functions like that `extractSyncSetting` to handle the job they need to do. 

I know that those 2 things are almost the same (just one is happening live with paired devices and another when needed), but I like clear separation like having a top level function for handling backed up messages (like this one [here](https://github.com/status-im/status-go/blob/5005b7b3cf45fed801257060fd68e19c4422e317/protocol/messenger_backup_handler.go#L21-L59) where we will add new things we need to handle) and having top level function for handling device syncing messages and they may call the same functions after that.

Reply:
> and we should invoke something like messageState.Response.AddSettings() to modify Response(@cammellos told me before in my old prs before), and it would be good within handleXXX, sorry I did not review your pr in time 😥 WDYT?

This showed up while I was writing my previous message and didn't see it before.
So I wouldn't mix signals for notifying client about syncing among devices and applied fetched data from waku.
That's why I introduced new signal type and sending signals only in case if data from waku are applied (means if there are newer than those stored in the db). Also according to UI design we have there is a fetching data progress signal which supports it.